### PR TITLE
Feature/tg2141 redux actions

### PR DESCRIPTION
**Pull request by Vegter**
Add meta data to actions:
- ignore (The action will not change the url)
- replace (The action will replace the url (not adding a new entry in the browser history))
- isButton (The action will be triggered by a button instead of a link)

This metadata is used by the reducers and dp-link
The following actions have been replaced by a dp-link button:
MAP_FULLSCREEN
HIDE_STRAATBEELD
SHOW_LAYER_SELECTION
HIDE_LAYER_SELECTION
HIDE_PRINT

### CODE REVIEW

File: modules/map/components/toggle-fullscreen/toggle-fullscreen.html [hunk 1]
```diff
@@ -1,9 +1,5 @@
...
-            'c-toggle-fullscreen__icon--minimize': vm.isFullscreen
-        }"
-        ng-click="vm.toggle()"
-        title="{{vm.buttonText}}">
-    <span class="u-sr-only">{{vm.buttonText}}</span>
-</button>
\ No newline at end of file
+<dp-link type="MAP_FULLSCREEN"
+         payload="!vm.isFullscreen"
+         class-name="{{'c-toggle-fullscreen ' + 'c-toggle-fullscreen__icon--' + (vm.isFullscreen ? 'minimize' : 'fullscreen')}}"
```
Comment (boris-arkenaar) [inline modules/map/components/toggle-fullscreen/toggle-fullscreen.html]: De eerste `' + '` zou je weg kunnen laten.

File: modules/shared/components/link/link.component.js [hunk 2]
```diff
@@ -24,32 +24,48 @@
         };
 
         function linkFn (scope, element) {
-            let debounced = debounce(300, update),
-                unsubscribe = store.subscribe(debounced);
+            const DEBOUNCE_WAIT_TIME = 300;     // Wait time in msecs before executing update
```
Comment (boris-arkenaar) [inline modules/shared/components/link/link.component.js]: Netjes!

File: modules/shared/components/link/link.component.test.js [hunk 3]
```diff
@@ -54,7 +55,10 @@ describe('The dp-link component', function () {
                 $provide.constant('ACTIONS', {
                     SHOW_PAGE: 'show-page',
                     MAP_PAN: 'map-pan',
-                    SHOW_LAYER_SELECTION: 'show-layer-selection'
+                    SHOW_LAYER_SELECTION: 'show-layer-selection',
+                    ISBUTTON: {
```
Comment (boris-arkenaar) [inline modules/shared/components/link/link.component.test.js]: `IS_BUTTON`

File: modules/shared/components/link/link.component.test.js [hunk 4]
```diff
@@ -103,6 +108,26 @@ describe('The dp-link component', function () {
...
 
+    it('can be a button', function () {
+        let component = getComponent('ISBUTTON', 'payload', 'className', 'hoverText');
+
+        expect(store.subscribe).not.toHaveBeenCalled();
+        expect(component.find('a').length).toBe(0);
+
+        let button = component.find('button');
+        expect(button.length).toBe(1);
+        expect(button.attr('title')).toBe('hoverText');
```
Comment (boris-arkenaar) [inline modules/shared/components/link/link.component.test.js]: Je zou ook nog kunnen testen of de `hoverText` gezet is in het `.u-sr-only` element.

File: modules/map/components/toggle-fullscreen/toggle-fullscreen.html (hunk 1 above)
Comment (Vegter) [inline modules/map/components/toggle-fullscreen/toggle-fullscreen.html]: Top, aangepast

File: modules/shared/components/link/link.component.js (hunk 2 above)
Comment (Vegter) [inline modules/shared/components/link/link.component.js]: Thanks

File: modules/shared/components/link/link.component.test.js (hunk 4 above)
Comment (Vegter) [inline modules/shared/components/link/link.component.test.js]: Top, aangepast

**Pull request merged**